src/python/torchdistx/optimizers/anyprecision_optimizer.py

```diff
@@ -8,15 +8,32 @@
 # with optional Kahan summation for high precision weight updates.
 # Allows direct control over momentum, variance and auxiliary compensation
 # buffer dtypes.
-# Optional Kahan summation is used to offset precision reduction for
-# the weight updates. This allows full training in BFloat16 (equal or
-# better than FP32 results in many cases) due to high precision weight upates.
+# Optional Kahan summation is used to enable high precision for
+# the weight updates. This allows sucessful training in pure BFloat16
+# (often equal or better than FP32 results) due to high precision weight
+# updates, while training with reduced GPU memory and
+# increased training speed.
 
 import torch
+import torch.cuda.nccl as nccl
+import torch.distributed as dist
+from pkg_resources import packaging
 from torch.optim.optimizer import Optimizer
 
 
 class AnyPrecisionAdamW(Optimizer):
+    def verify_bfloat_support(
```

Comment:
_verify_bfloat_support as its not a public API. 

Also conventionally, all methods usually come after __init__ definition.

Reply:
updated to _, and have moved it so init is first!  Thanks for the feedback!